osmnames/import_osm/set_names.sql

```diff
@@ -0,0 +1,52 @@
+-- If the 'name' is empty or null, get name in another available language
+-- Order of priority for 'name' is: default name, en, fr, de, es, ru, zh
+DROP FUNCTION IF EXISTS get_name(HSTORE);
+CREATE FUNCTION get_name(all_tags HSTORE) RETURNS TEXT AS $$
+  SELECT COALESCE(all_tags -> 'name:en',
+                  all_tags -> 'name:fr',
+                  all_tags -> 'name:de',
+                  all_tags -> 'name:es',
+                  all_tags -> 'name:ru',
+                  all_tags -> 'name:zh');
+$$ LANGUAGE 'sql' IMMUTABLE;
+
+
+
+CREATE OR REPLACE FUNCTION array_distinct(anyarray)
+RETURNS anyarray AS $$
+  SELECT ARRAY(SELECT DISTINCT unnest($1))
+$$ LANGUAGE sql;
+
+
+/*  Get values for all the name tags as defined in http://wiki.openstreetmap.org/wiki/Key:name
+    The alternative_names string will contain all distinct names except the one already set as 'name' */
+DROP FUNCTION IF EXISTS get_alternative_names(HSTORE, TEXT, VARCHAR);
```

Comment:
additionally i wouldn't provide the delimiter as a function parameter, since it is always the same. I would hard code it here...